packages/graphql/lib/schema-builder/storages/type-metadata.storage.ts

```diff
@@ -216,20 +216,30 @@ export class TypeMetadataStorageHost {
   }
 
   loadClassPluginMetadata(metadata: ClassMetadata[]) {
+    const seen = new Set<Function>();
     metadata
       .filter((item) => item?.target)
-      .forEach((item) => this.applyPluginMetadata(item.target.prototype));
+      .forEach((item) => this.applyPluginMetadata(item.target.prototype, seen));
   }
 
-  applyPluginMetadata(prototype: Function) {
+  applyPluginMetadata(prototype: Function, seen = new Set<Function>()) {
     do {
       if (!prototype.constructor) {
         return;
       }
-      if (!prototype.constructor[METADATA_FACTORY_NAME]) {
+      // Skip redundant metadata merges for common parents.
+      if (seen.has(prototype.constructor)) {
+        return;
+      }
+      seen.add(prototype.constructor);
+
+      const metadata = Object.getOwnPropertyDescriptor(
```

Comment:
question: any specific reason why we use `getOwnPropertyDescriptor` now?

Reply:
Yes this is basically the fix. We only want the metadata factory from the referenced class. Static properties are inherited, and we want to avoid this here.
This could be an `Object.hasOwn` check. I used this to do that check and get the _own_ value at the same time so the constant `METADATA_FACTORY_NAME` only had to be referenced once.

Reply:
I clarified the PR description. The minimal reproduction is still a lot of boilerplate.
I looked at updating tests, but I didn't find a spot where I could using the plugin metadata _and_ instantiate a `GraphQLModule` using it.
What would you like me to do?